dv/cosim/spike_cosim.cc

```diff
@@ -571,11 +571,59 @@ void SpikeCosim::initial_proc_setup(uint32_t start_pc, uint32_t start_mtvec,
 }
 
 void SpikeCosim::set_mip(uint32_t mip) {
+  uint32_t old_mip;
+  uint32_t new_mip;
+
+  new_mip = mip;
+  old_mip = processor->get_state()->mip->read();
```

Comment:
Since this is C++ (and not ANSI C or (System)Verilog), combining declaration and value assignment is probably preferable:
```cpp
uint32_t new_mip = mip;
uint32_t old_mip = processor->get_state()->mip->read();
```
(suggestion also applies to other C++ variables added in this PR)